File: transaction-bff/src/main/java/br/com/felipe/gadelha/coffeandit/transactionbff/api/v1/dto/request/AccountRq.java
package br.com.felipe.gadelha.coffeandit.transactionbff.api.v1.dto.request;

import br.com.felipe.gadelha.coffeandit.transactionbff.domain.entity.Account;

public class AccountRq {
    private final Long id;
    private final Long agency;
    public AccountRq(Account account) {
        this.id = account.getId();
        this.agency = account.getAgency();
    }
    public Account toEntity() {
        return new Account(id, agency);
    }
}

File: transaction-ms/src/main/kotlin/br/com/felipe/gadelha/coffeandit/transactionms/api/v1/dto/request/AccountRq.kt
package br.com.felipe.gadelha.coffeandit.transactionms.api.v1.dto.request

import br.com.felipe.gadelha.coffeandit.transactionms.domain.entity.Account

class AccountRq(
    private val id: Long,
    private val agency: Long
) {

    fun toEntity(): Account {
        return Account(id, agency.toString())
    }
}
